config.go

```diff
@@ -321,6 +333,41 @@ func validateVolumes(volumes Volumes) (errors []string) {
 	return
 }
 
+// ValidateBuildEnvs checks that input BuildEnvs are correct and contain all necessary fields.
+// Returns array of error messages, empty if no errors are found
+//
+// Allowed settings:
+// - name: EXAMPLE1                					# ENV directly from a value
+//   value: value1
+// - name: EXAMPLE2                 				# ENV from the local ENV var
+//   value: {{ env:MY_ENV }}
+func ValidateBuildEnvs(envs Envs) (errors []string) {
+	for i, env := range envs {
+		if env.Name == nil && env.Value == nil {
+			errors = append(errors, fmt.Sprintf("env entry #%d is not properly set", i))
+		} else if env.Value == nil {
+			errors = append(errors, fmt.Sprintf("env entry #%d is missing value field, only name '%s' is set", i, *env.Name))
+		} else {
+
+			if err := utils.ValidateEnvVarName(*env.Name); err != nil {
+				errors = append(errors, fmt.Sprintf("env entry #%d has invalid name set: %q; %s", i, *env.Name, err.Error()))
+			}
+
+			if strings.HasPrefix(*env.Value, "{{") {
+				// ENV from the local ENV var; {{ env:MY_ENV }}
+				if !regLocalEnv.MatchString(*env.Value) {
+					errors = append(errors,
+						fmt.Sprintf(
+							"env entry #%d with name '%s' has invalid value field set, it has '%s', but allowed is only '{{ env:MY_ENV }}', '{{ secret:secretName:key }}' or '{{ configMap:configMapName:key }}'",
```

Comment:
this part is not relevant here `'{{ secret:secretName:key }}' or '{{ configMap:configMapName:key }}'`

Reply:
Yeah, we can add it here then :)